dos/src/main.rs

```diff
@@ -504,17 +566,59 @@ fn main() {
     }
 
     info!("done found {} nodes", nodes.len());
-    run_dos(&nodes, 0, cmd_params);
+
+    // create client which is used for airdrop and get blockhash
+    let client = if cmd_params.transaction_params.valid_blockhash {
+        let num_nodes = 1;
+        info!("Connecting to the cluster");
+        let nodes = discover_cluster(
```

Comment:
This ignores the `--skip-gossip` flag, not to mention has already been resolved above in the case where `--skip-gossip` has not been passed.  Just use of `_validators` from the `discover()` call

Reply:
Agree, I've got rid of extra code in https://github.com/solana-labs/solana/pull/24412/commits/d8c6dd2bba6be95dfbe967db80c87e10d307c9bb